components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1888,6 +1888,30 @@ void RewardsServiceImpl::SaveRecurringTip(const std::string& publisher_key,
                                       AsWeakPtr(), std::move(callback)));
 }
 
+void RewardsServiceImpl::SetMonthlyContribution(
+    const std::string& publisher_key,
+    double amount,
+    base::OnceCallback<void(bool)> callback) {
+  if (!Connected()) {
+    return DeferCallback(FROM_HERE, std::move(callback), false);
+  }
+
+  auto on_completed = [](base::WeakPtr<RewardsServiceImpl> self,
```

Comment:
Can we use a member function (`OnSetMonthlyContribution()`) instead?

Reply:
Done.